thavalon-server/src/game/messages.rs

```diff
@@ -128,6 +128,21 @@ pub enum Message {
         winning_team: Team,
         roles: HashMap<String, RoleDetails>,
     },
+
+    /// Message that a client should surface to the end user.
+    Toast {
+        severity: ToastSeverity,
+        message: String
+    }
+}
+
+/// Severity of a toast notification
+#[derive(Debug, Clone, Eq, PartialEq, Serialize)]
+#[serde(tag = "toastSeverity")]
+pub enum ToastSeverity {
```

Comment:
Do we need Debug, Clone, Eq, and PartialEq on this? `Clone` allows the `.clone()` function to work, and `Eq` and `PartialEq` are related to using the enum in equality comparison expressions.

Reply:
We can also remove the `#[serde(tag = "toastSeverity")]`. That changes the emitted JSON to a format of "toastSeverity: <severity>'. Since we already have a severity field though, we don't need the extra type and can just output the name directly.

Reply:
We need all of those because a ToastSeverity is included in a Message, which has all of those tags and requires the contained types to implement them as well.

good to know about the tag, I wasn't sure whether the default was no formatting or just breaking. neat!